framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,202 @@ private void makeGroundTargetType(
+
+        AnnotatedDeclaredType typeDeclaration =
+                (AnnotatedDeclaredType)
+                        getAnnotatedType(typeToCapture.getUnderlyingType().asElement());
+
+        // Create a mapping from type variables in typeDeclaration to type arguments.
+        Map<TypeVariable, AnnotatedTypeMirror> typeVarToTypeArguments = new HashMap<>();
+        Map<TypeVariable, AnnotatedTypeMirror> captureToAnnotatedCapture = new HashMap<>();
+        for (int i = 0; i < typeDeclaration.getTypeArguments().size(); i++) {
+            TypeVariable typeVar =
+                    (TypeVariable) typeDeclaration.getTypeArguments().get(i).getUnderlyingType();
+            AnnotatedTypeMirror preCapturedArg = typeToCapture.getTypeArguments().get(i);
+            AnnotatedTypeMirror capturedTypeArg = capturedType.getTypeArguments().get(i);
+            if (TypesUtils.isCaptured(capturedTypeArg.getUnderlyingType())
+                    && preCapturedArg.getKind() == TypeKind.WILDCARD) {
+                // The type argument is a captured type. Use the type argument from the newly
+                // created and yet-to-be annotated captureType. (The annotations are added as part
+                // of capturing the wildcard.)
+                typeVarToTypeArguments.put(typeVar, capturedTypeArg);
+                // Also, add a mapping from the captured type mirror to the annotated captured
+                // type, so that if one captured type refers to another, the correct annotated type
+                // is used.
+                captureToAnnotatedCapture.put(
+                        ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+                        capturedTypeArg);
+            } else {
+                // The type argument is not a captured type. Use the type argument from
+                // typeToCapture, which is fully-annotated.
+                typeVarToTypeArguments.put(typeVar, preCapturedArg);
+            }
+        }
+
+        // Use the mapping above to substitute the type variables in capturedType and
+        // typeDeclaration.
+        capturedType =
+                (AnnotatedDeclaredType)
+                        typeVarSubstitutor.substituteWithoutCopyingTypeArguments(
+                                typeVarToTypeArguments, capturedType);
+
+        // Loop through the type arguments and set the annotations of each captured type.
+        List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>(typeToCapture.typeArgs);
+        List<TypeVariable> orderToCapture =
+                TypesUtils.order(captureToAnnotatedCapture.keySet(), types);
+        for (TypeVariable capture : orderToCapture) {
+            AnnotatedTypeMirror typeArg = captureToAnnotatedCapture.get(capture);
+            int i = capturedTypeMirror.getTypeArguments().indexOf(capture);
+            AnnotatedTypeMirror annoTypeArg = typeToCapture.getTypeArguments().get(i);
```

Comment:
What is the distinction between `typeArg` and `annoTypeArg`?  Both are `AnnotatedTypeMirror`s.  Their names only reference their type, not their semantics or content.  Can you rename them?